From 5d5b2b6c9bf89c05d368d34fd2770f58eef6826e Mon Sep 17 00:00:00 2001
From: Alex Naidis <alex.naidis@linux.com>
Date: Sun, 21 Aug 2016 18:24:22 +0200
Subject: [PATCH 4/6] aapt: allow opting out of uncompressed aapt

* extends this commit: cdcd4fbea499aba637a142ad6f4b083ce7f1fcc2
* arter97 reported that some devices with small /system partitions
  can get issues.
  -> create an option to opt out

To opt out, one just has to specify "TARGET_WANTS_AAPT_COMPRESS := true"
in the device's Boardconfig.

Change-Id: If9e7b07ed39f3d180b284a96e3e05903349afc9e
Signed-off-by: Alex Naidis <alex.naidis@linux.com>
Signed-off-by: Park Ju Hyung <qkrwngud825@gmail.com>
---
 tools/aapt/Android.mk | 4 ++++
 tools/aapt/Main.cpp   | 6 +++++-
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/tools/aapt/Android.mk b/tools/aapt/Android.mk
index 04f46d9b27f..250fffd8eb3 100644
--- a/tools/aapt/Android.mk
+++ b/tools/aapt/Android.mk
@@ -67,6 +67,10 @@ aaptHostStaticLibs := \
 aaptCFlags := -DAAPT_VERSION=\"$(BUILD_NUMBER_FROM_FILE)\"
 aaptCFlags += -Wall -Werror
 
+ifeq ($(TARGET_WANTS_AAPT_COMPRESS),true)
+aaptCFlags += -DAAPT_COMPRESS=1
+endif
+
 aaptHostLdLibs_linux := -lrt -ldl -lpthread
 
 # Statically link libz for MinGW (Win SDK under Linux),
diff --git a/tools/aapt/Main.cpp b/tools/aapt/Main.cpp
index 193d6b24b21..6019f76afdf 100644
--- a/tools/aapt/Main.cpp
+++ b/tools/aapt/Main.cpp
@@ -268,8 +268,12 @@ int main(int argc, char* const argv[])
     int result = 1;    // pessimistically assume an error.
     int tolerance = 0;
 
-    /* default to 0 compression */
+    /* default to 0 compression, unless the target opted out*/
+#if AAPT_COMPRESS
+    bundle.setCompressionMethod(ZipEntry::kCompressDeflated);
+#else
     bundle.setCompressionMethod(ZipEntry::kCompressStored);
+#endif
 
     if (argc < 2) {
         wantUsage = true;
-- 
2.11.0

